build/app/webpack.js

```diff
@@ -43,33 +39,48 @@ const baseConfig = require(path.join(process.cwd(), 'n-ui-build.config.js'));
 
 noGitignoreWildcard();
 
-// we no longer build a main.js for the app when generating the standard asset variants
-const variants = [
-	// all entry points excluding main.js generated as normal
-	headCss(nWebpack(Object.assign({}, baseConfig, {
-		entry: filterEntryKeys(baseConfig.entry, /main\.js$/, true)
-	})))
-]
-
-// new entry point for main.js declaring external n-ui
-const mainJs = nWebpack(Object.assign(clone(baseConfig), {
-	language: 'js',
-	entry: modifyEntryKeys(baseConfig.entry, /main\.js$/, name => name.replace(/\.js$/,'-without-n-ui.js'))
-}))
-
-const nUiEntry = path.join(process.cwd(), 'bower_components/n-ui/browser/js/webpack-entry');
-const nUiEntryPoints = require(nUiEntry)(baseConfig.nUiExcludes)
-mainJs.externals = Object.assign({}, mainJs.externals, nUiEntryPoints);
-mainJs.plugins.push(
+const webpackConfigs = [];
+
+/*
+We no longer build a main.js for the app when generating the standard asset variants
+so this config is for all entry points defined by an app *excluding* the main.js one
+
+Mostly this config will only be for main.css.
+*/
+const nonMainJsWebpackConfig = nWebpack();
+nonMainJsWebpackConfig.entry = filterEntryKeys(baseConfig.entry, /main\.js$/, true);
```

Comment:
you could just `delete entry['main.js']` if we're thinking of moving down the mroe standardised approach